### PR TITLE
feat(chrome_extension): write dom changes to database (using sockets) during openadapt.record

### PR DESCRIPTION
**Pull request by KrishPatel13**
**What kind of change does this PR introduce?**
This PR implements the chrome extension to write the DOM changes (documents' outer HTML of Head and Body and the url) in database.

**Summary**
Motivation: Related Issue: https://github.com/OpenAdaptAI/OpenAdapt/issues/51

**Checklist**
* [ ] My code follows the style guidelines of OpenAdapt
* [ ] I have performed a self-review of my code
* [ ] If applicable, I have added tests to prove my fix is functional/effective
* [ ] I have linted my code locally prior to submission
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [ ] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

**Other information**

### CODE REVIEW

File: openadapt/models.py [hunk 1]
```diff
@@ -323,3 +323,11 @@ class MemoryStat(db.Base):
...
     memory_usage_bytes = sa.Column(ForceFloat)
     timestamp = sa.Column(ForceFloat)
+
+
+class BrowserEvent(db.Base):
+
+    bodyHTML = sa.Column(sa.String)
+    headHTML = sa.Column(sa.String)
+    url = sa.Column(sa.String)
+    timestamp = sa.Column(ForceFloat)
```
Comment (abrichr) [inline openadapt/models.py:333]: Please add `recording_timestamp`

Comment (KrishPatel13) [reply openadapt/models.py]: Addressed in: [1aade8e](https://github.com/OpenAdaptAI/OpenAdapt/pull/364/commits/1aade8e03a15fe9f3944252915454760faac1eac) and [ba220be](https://github.com/OpenAdaptAI/OpenAdapt/pull/364/commits/ba220be409484d871336c71d9b5468a2f1553484)

File: native_chrome_extension/browser.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        received_message = get_message()
+
+        # Sending message to Client
+        sockets.server_send_message(config.SOCKET_PORT, received_message)
+
+        # # Receiving message from Server
+        # response = sockets.server_receive_message(config.SOCKET_PORT)
+
+        # Sending the received message back to background.js
+        send_message(encode_message(received_message))
```
Comment (0dm) [inline native_chrome_extension/browser.py:99]:
```suggestion
        send_message(encode_message(received_message))
        
```
newline

Comment (KrishPatel13) [reply native_chrome_extension/browser.py:103]: [11bce46](https://github.com/OpenAdaptAI/OpenAdapt/pull/364/commits/11bce463ffcac14a446653896e1f18f602f5c98e)

**Comment by KrishPatel13** [comment]
This the point where the sockets.py can be integrated into the record.py

**Comment by abrichr** [comment]
@0dm are you interested in picking this up? I think it's quite close to being done.

**Comment by 0dm** [comment]
@abrichr Sure.

**Comment by KrishPatel13** [comment]
> currently have these two messages after trying to load unpacked extension on chrome: image

The first one is cause ue to the fact that the The extension was not able to connect to the Native Messaging Host (app). The common reason for this error is that the naming of the native messaging host is not consistent with the files, esp. manifest.json, browser.sh, native_manifest.json etc.

The second one is a common one. I got that quite a lot of times. That is basically given when you try to use the `port` variable when is it None. Generally the port var is None when is was not able to connect to the NativemessagingHost.

**Comment by abrichr** [comment]
This suggests something wrong with `native_chrome_extension/browser.bat`.

@0dm can you please push `native_chrome_extension/browser.sh` ?

**Comment by 0dm** [comment]
sure I'll push it

**Comment by abrichr** [comment]
Also how are you handling `path` in `native_chrome_extension/openadapt_chrome.json`?

**Comment by 0dm** [comment]
it's an absolute path to the script : users/aaron/Documents/GitHub/OpenAdapt/native_chrome_extension/

**Comment by 0dm** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=1468564 posted here

**Comment by abrichr** [comment]
@0dm can you please try running `lsof` or similar to see what processes are running, and what files/ports they are accessing?

**Comment by abrichr** [comment]
@0dm @KrishPatel13 can you please remind me what is the status on this?

File: chrome/browser.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+import json
+import sqlite3
+import struct
+import sys
+
+from openadapt import config, sockets
+
+SOCKETS = True
+DBG_DATABASE = False
```
Comment (abrichr) [inline chrome/browser.py:20]: Can you please clarify what this is for?

File: chrome/browser.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+# than text, mode.
+
+import json
+import sqlite3
+import struct
+import sys
+
+from openadapt import config, sockets
+
+SOCKETS = True
```
Comment (abrichr) [inline chrome/browser.py:19]: Will this ever be False?

**Comment by abrichr** [comment]
@KrishPatel13 I believe this should be closed in favor of https://github.com/OpenAdaptAI/OpenAdapt/pull/744, can you please confirm?

**Comment by KrishPatel13** [comment]
> @KrishPatel13 I believe this should be closed in favor of https://github.com/OpenAdaptAI/OpenAdapt/pull/744, can you please confirm?

Yes!

**Pull request closed without merging**